### ISSUE TITLE
Generating xml documentation causes `type not found` error

### ISSUE THREAD

**Issue by tienery** [Kode/Kha#486]
I am trying to generate documentation using `dox` for my library, Twinspire, but seem to be getting an error when using the following hxml:

```hxml
-cp twinspire/
-D doc-gen
-lib kha
--macro include('kha')
--macro include('twinspire')
--no-output
--each

--next
-xml docs/twinspire-js.xml
-js dummy.js
```

This is the error:

    twinspire/events/Event.hx:13: characters 7-14 : Type not found : kha.Key

This does not, however, happen if I generate documentation using a proper Kha project, using the following `khafile.js`:

```js
let project = new Project('Tutorial');
project.addAssets('Assets/**');
project.addSources('Sources');
project.addLibrary('twinspire');
project.addLibrary('dox');
project.addDefine('--macro include("kha")');
project.addDefine('--macro include("twinspire")');
project.addParameter('-xml docs/twinspire.xml');
resolve(project);
```

Of course, the above only works on Types you `import` into the project, which means its less effective than the former to generate documentation. I know this may not specifically be a Kha-based issue, but I could not think of a better place to post this problem.

Any ideas?

**Comment by RobDangerous** [comment]
Kha is not a lib, -lib generally does not work. But you can run khamake, then grab the generated hxml file in the build subdir and work from there.

**Comment by tienery** [comment]
Hmm... I think I tried this approach, but in a different. I ended up with tonnes of errors in Kha, because I don't think the code-base is probably designed with documentation generation in mind. For example, there are `extern` classes that do not have an `@:native` meta-data, causing errors such as `F:\Github\Kha\Sources/kha/compute/Compute.hx:6: characters 15-75 : Type required for extern classes and interfaces`.

Some classes aren't wrapped with `#if sys` conditional statements, meaning that those classes that rely on system calls won't work in JavaScript. I've even had to rewrite a lot of things in `CssPainter.hx` in the js backend to get documents generated.

I gave up on it by the time I reached the error above. There is probably a better way to handle these issues, but I wasn't sure. That's why I posted here.

Is there alternative documentation generators that does not rely on the Haxe compiler, by any chance? Or are they all like that?
